chapter.md

```diff
@@ -0,0 +1,735 @@
+
+```js
+let state = {color: 'red'};
+```
+
+and the reducer cycles between colors:
+
+```js
+function changeColor(state, action) {
+  switch(action.type) {
+
+  case 'NEXT':
+    if (state.color == 'red')
+      return {color: 'green'};
+    else if (state.color == 'green')
+      return {color: 'amber'};
+    else if (state.color == 'amber')
+      return {color: 'green'};
+
+  default:
+    return state;
+  }
+}
+```
+
+It's very important that `changeColor` returns the original state without modification
+when it doesn't recognize the type of the action it is being asked to perform,
+since this allows us to combine reducer functions
+without worrying that they will trip over one another.
+And to simplify our program,
+we frequently fold the initialization of the state into the definition of the reducer
+by defining the initial state as the default value for the reducer's first parameter:
+
+```js
+function changeColor(state = {color: 'red'}, action) {
+  switch(action.type) {
+    ...as before...
+  }
+}
+```
+
+Once this reducer is defined,
+our application wraps it up to create an object store,
+and then sends actions to that store to tell it when to move to the next state:
+
+```js
+import { createStore } from 'redux';
+let store = createStore(changeColor);
+
+for (let i=0; i<10; i++) {
+  store.dispatch({type: 'NEXT'});
+  console.log(store.getState().color); // red, green, amber, red, ...
+}
+```
+
+If we want to add new capabilities,
+we simply update our reducer:
+
+```js
+function changeColor(state, action) {
+  switch(action.type) {
+
+    case 'NEXT':
+      ...as above...
+
+    case 'EMERGENCY':
+      return {color: 'red'};
+
+    default:
+      return state;
+  }
+}
+```
+
+This may seem like a lot of work to manage a single traffic light,
+but that work pays off as soon as we have to worry about asynchronous updates.
+For example,
+if we want to do an emergency test of the light every night at 1:00 am,
+all we have to do is this:
+
+```js
+let delay = until(tomorrow() + ONE_HOUR);
+setTimeout(() => changeColor(state, {type: 'EMERGENCY'}), delay);
+```
+
+When the timeout callback is triggered,
+Redux will put the traffic light in the required state
+regardless of what else has gone on or is going on.
+
+## Setting Up Forms to Work with Redux
+
+As our running example,
+we will build a set of forms that allow users to create a wizard for a role-playing game.
+This example was built by Renee Vrantsidis and Daniel Figueiredo
+for [the talk][repo-talk] this chapter is based on,
+and the complete application is [available on GitHub][repo-application].
+
+The first step is to plan the shape of our forms.
+We can structure the store however we want,
+but since we know that real applications evolve,
+we will define things using TypeScript interfaces
+rather than relying on specific concrete classes,
+since that will make it easier to swap things in and out for testing later on.
```

Comment:
I like you're explaining briefly why we're using interfaces instead of classes `rather than relying on specific concrete classes`
Not sure if it fits in here but maybe add something saying that it also helps avoiding the `this` keyword usages and helps enforcing your things to be stateless, more pure functional